docs/public/_redirects

```diff
@@ -1,16 +1,23 @@
-# Proxies
+/ /x/advanced-components/
 
-## mui-x is configured to be hosted at the root.
-/static/* https://material-ui-docs.netlify.app/static/:splat 200
+# For links that we can't edit later on, e.g. hosted in the code published on npm or sent by email
+# should all be prefixed with x-
+/r/x-license https://mui.com/store/items/material-ui-pro/ 302
+/r/x-license-eula https://mui.com/store/legal/mui-x-eula/ 302
+/r/x-license-key-installation https://mui.com/x/advanced-components/#license-key-installation 302
 
-## Localization
+# Legacy redirection
+# 2021
 /fr/* /:splat 200
 /de/* /:splat 200
 /ja/* /:splat 200
+# 2022
+/x/license/ https://mui.com/store/legal/mui-x-eula/ 301
+/x/react-data-grid/group-pivot/ /x/react-data-grid/tree-data/ 301
```

Comment:
Yes, my assumption is that 404 is worse than not linking the exact correct content.

For the client side fragment redirections, why not, we never implemented this before it could help, but from an SEO perspective, I doubt that Google will ever be able to update it's index from it, so we still need to server side redirect to a working page.